### PR TITLE
Adding TLS communication for setup_etcd

### PR DESCRIPTION
**Pull request by vibhorkum**
This commit adds etcd communication in ETCD

### CODE REVIEW

**Review by jt-edb — APPROVED** [review]
LGTM

File: roles/setup_etcd/tasks/setup_etcd.yml [hunk 1]
```diff
@@ -64,7 +68,7 @@
 - name: Prepare etcd_inital_cluster based on private_ip
   ansible.builtin.set_fact:
     etcd_initial_cluster: >-
-      {{ etcd_initial_cluster | default([]) + [node.inventory_hostname + '=http://' + node.private_ip + ':' + etcd_peer_port | string] }}
+      {{ etcd_initial_cluster | default([]) + [node.inventory_hostname + http_type + node.private_ip + ':' + etcd_peer_port | string] }}
```
Comment (hannahms) [inline roles/setup_etcd/tasks/setup_etcd.yml:71]: Should this also be `_http_type`?

**Review by hannahms — APPROVED** [review]
Double check the two comments made, otherwise LGTM

File: roles/setup_etcd/tasks/setup_etcd.yml (hunk 1 above)
Comment (vibhorkumar123) [reply roles/setup_etcd/tasks/setup_etcd.yml:71]: Yes. I cannot see this in PR. Maybe you are looking at the old commit.

File: roles/setup_etcd/tasks/etcd_ssl_member_key.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    cmd: >-
+      openssl x509 -req \
+        -in {{ etcd_member_csr }} \
+        -days {{ etcd_ssl_valid_days }} \
+        -CA {{ etcd_ca_crt }} \
+        -CAkey {{ etcd_ca_key }} \
+        -CAcreateserial \
+        -extensions ssl_client \
+        -extfile {{ etcd_ssl_dir }}/openssl.conf \
+        -days {{ etcd_ssl_valid_days }} \
```
[the remark this answers is not in the document]
Comment (vibhorkumar123) [reply roles/setup_etcd/tasks/etcd_ssl_member_key.yml]: Good catch. Fixed.

File: roles/setup_etcd/tasks/setup_etcd.yml (hunk 1 above)
Comment (vibhorkumar123) [reply roles/setup_etcd/tasks/setup_etcd.yml]: Nevermind. Found it. Fixed.

**Pull request merged**